src/components/encoding-pane/function-picker.tsx

```diff
@@ -43,7 +61,12 @@ export class FunctionPickerBase extends React.PureComponent<FunctionPickerProps,
     }
   }
   private onFunctionChange(event: any) {
-    this.props.onFunctionChange(event.target.value as ShelfFunction);
+    const {index} = this.props;
+    if (index !== undefined) {
+      this.props.onFunctionChange(event.target.value, index);
```

Comment:
Yea, I could extract single filter shelf as one file, so that I can read index as a prop, just like the encoding shelf. @kanitw Do you think this is worth doing?